src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentAction.java

```diff
@@ -617,6 +625,13 @@ public ActionForward route(ActionMapping mapping, ActionForm form, HttpServletRe
             added &= addNewItem(iWantDocForm, iWantDocument, item);
         }
         
+        if (added && iWantDocument.getFavoriteAccountLineIdentifier() != null) {
+            added &= addNewFavoriteAccount(iWantDocument);
+            if (added) {
+                iWantDocument.setFavoriteAccountLineIdentifier(null);
+            }
+        }
+        
```

Comment:
This block is called three times here, do we have an opportunity to extract it to a function to eliminate duplication?